tokio/src/macros/cfg.rs

```diff
@@ -364,6 +364,16 @@ macro_rules! cfg_unstable {
     }
 }
 
+macro_rules! cfg_trace {
+    ($($item:item)*) => {
+        $(
+            #[cfg(all(feature = "tracing", tokio_unstable))]
```

Comment:
We talked about this, but I don't think `tracing` merits being placed behind the `tokio_unstable` unstable flag. @carllerche will need to weigh in on this, but unlike the `CancelationToken`, changes to the `tracing` output aren't what the Tokio project has historically considered to be a "breaking change". An off-by-default (e.g., excluded from `full`) feature flag seems like a reasonable default.

Reply:
I think at one point, @carllerche suggested that any dependency on a pre-1.0 crate should require the `tokio_unstable` flag...but, in this case, we aren't exposing anything from `tracing` in Tokio's public APIs. AFAICT, it shouldn't be possible for the `tracing` dependency to cause any compile time breakage --- the worst failure mode, if there is eventually a breaking change to the `tracing-core` crate that we can't work around, is that the tracing diagnostics might silently fail to be collected. 

IMO, you're right, and an off-by-default optional dep should be fine. That would be pretty similar to [what we do for `parking_lot`](https://github.com/tokio-rs/tokio/blob/f69e5bfb87eda2b8de9b762b9280f1733cdaa096/tokio/Cargo.toml#L107), which is also pre-1.0 (and churns versions somewhat frequently!), since it's also not exposed publicly. Of course, if we _were_ to expose `tracing` types (such as `Span` or `Dispatch`) in a public API later on, _those_ APIs should require the unstable flag.

With all of that said: I made the more conservative choice of opening this PR _with_ the `tokio_unstable` flag required, because I want to bias towards getting this merged sooner rather than later, and continuing to iterate once it merges. We can always remove the flag in a follow-up, but it's harder to _add_ if these changes ship in a published release...